pandas/core/arrays/string_.py

```diff
@@ -73,51 +81,134 @@ class StringDtype(ExtensionDtype):
+            )
+        if storage == "pyarrow" and pa_version_under1p0:
+            raise ImportError(
+                "pyarrow>=1.0.0 is required for PyArrow backed StringArray."
+            )
+
+        self.storage = storage
+
+    @property
+    def name(self):
+        return f"string[{self.storage}]"
 
     @property
     def type(self) -> type[str]:
         return str
 
     @classmethod
-    def construct_array_type(cls) -> type_t[StringArray]:
+    def construct_from_string(cls, string):
+        """
+        Construct a StringDtype from a string.
+
+        Parameters
+        ----------
+        string : str
+            The type of the name. The storage type will be taking from `string`.
+            Valid options and their storage types are
+
+            ========================== ==============================================
+            string                     result storage
+            ========================== ==============================================
+            ``'string'``               pd.options.mode.string_storage, default python
+            ``'string[python]'``       python
+            ``'string[pyarrow]'``      pyarrow
+            ========================== ==============================================
+
+        Returns
+        -------
+        StringDtype
+
+        Raise
+        -----
+        TypeError
+            If the string is not a valid option.
+
+        """
+        if not isinstance(string, str):
+            raise TypeError(
+                f"'construct_from_string' expects a string, got {type(string)}"
+            )
+        if string == "string":
+            return cls()
+        elif string == "string[python]":
+            return cls(storage="python")
+        elif string == "string[pyarrow]":
+            return cls(storage="pyarrow")
+        else:
+            raise TypeError(f"Cannot construct a '{cls.__name__}' from '{string}'")
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, str) and other == "string":
+            return True
+        return super().__eq__(other)
+
+    def __hash__(self) -> int:
+        # custom __eq__ so have to override __hash__
+        return super().__hash__()
+
+    # https://github.com/pandas-dev/pandas/issues/36126
+    # error: Signature of "construct_array_type" incompatible with supertype
+    # "ExtensionDtype"
+    def construct_array_type(  # type: ignore[override]
+        self,
+    ) -> type_t[StringArray | ArrowStringArray]:
         """
         Return the array type associated with this dtype.
 
         Returns
         -------
         type
         """
-        return StringArray
+        from pandas.core.arrays.string_arrow import ArrowStringArray
+
+        if self.storage == "python":
+            return StringArray
+        else:
+            return ArrowStringArray
 
-    def __repr__(self) -> str:
-        return "StringDtype"
+    def __repr__(self):
+        return self.name
```

Comment:
i think this would be fine, e.g. `string` is de-facto `string[python]`, though this means we can essentially *never* change it and `string[pyarrow]` for non-default storage types. maybe `string[object]` is more instructive?

Reply:
hmm, what about the global setting. the pyarrow backed string array should not be a second class citizen. At the moment if the global setting is changed , all tests pass.

Reply:
> `string` is de-facto `string[python]`

What `"string"` means depends on the global setting, which is *by default* (and not de-facto) `"string[python]"`. 

To be clear, I didn't want to suggest that "string" would always mean "string[python]". I just think that in the str representation (eg used in `df.dtypes`), we shouldn't show that detail.